Core/Src/main.c

```diff
@@ -87,18 +89,53 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+    //Initializing buffers
+    uint8_t transmit_bits[3] = {0x0, 0x1, 0x80}; // populated with start and configure bits for CH0
+    uint8_t receive_bits[3] = {0x0};
+
+    //Initializing constants and variables
+    const uint16_t MAX_ADC_VALUE = 0x400 - 1; //as it's a 10 bit ADC (represents 00000011 11111111)
```

Comment:
I appreciate the effort to make it HEX, but for a value like this, it would be more readable in the decimal numbering system, especially since you are going to be dividing with it later.